src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -379,6 +379,7 @@ public sealed partial class JsonSerializerOptions
         public bool IsReadOnly { get { throw null; } }
         public int MaxDepth { get { throw null; } set { } }
         public System.Text.Json.Serialization.JsonNumberHandling NumberHandling { get { throw null; } set { } }
+        public System.Text.Json.Serialization.JsonObjectCreationHandling PreferredObjectCreationHandling { get { throw null; } set { } }
```

Comment:
correct:
- preferred - means that if property's converter is capable of supporting populate then we will populate
- no preferred in name - we will use that or throw

now we will use setting in following order:
- property: no prefer here, we use it when it's non-null. If set incorrectly we throw
- type: prefer here: type properties will use this when they don't explicitly specify and their converter supports it
- options: preferred in name, if not other preference defined and property didn't define we will use this setting. Note it's non-nullable and Replace by default. We also don't enable it when converter doesn't support it.

Reply:
for number handling it's implied you have number so it's always supported as long as it's a number. Here it's a bit more fuzzy and you might not get what you want (i.e. immutable collections users might think it will append or it won't populate for structs if it doesn't have a setter). We've discussed that during API review and prefix was added during it